espnet/nets/pytorch_backend/transducer/blocks.py

```diff
@@ -526,11 +541,15 @@ def build_blocks(
         if block_type == "tdnn":
             module = build_tdnn_block(blocks_arch[i])
         elif block_type == "transformer":
+            enc_win_left = blocks_arch[i].get("win_left", 0)
```

Comment:
If `enc_win_left` and `dec_win_left` are block parameters, you should get values within build_transformer_block! However I think it should be a global parameter passed to `build_blocks` as we won't change window between decoder blocks? Also, as we can mix blocks, some checking should be done I guess

Reply:
1. Any suggestion for " get values within build_transformer_block"?
2. these window is for restrict the attention in encoder, I don't decoder attention need it. 
3. I will add some checks anyway, but I am not sure whether it satisfied your suggestion

Reply:
1. It's an optional parameter so you can follow same pattern as dropout parameters:
e.g.: `enc_win_left = block_arch["enc_win_left"] if "enc_win_left" in block_arch else 0.0`
2. Yes, not sure what you implied? You already used `net_part` to select attention class accordly
3. No problem! My suggestion is to define a `sliding_window` with left-right values in `asr_recog.py `to pass to `build_blocks`. If you see any usage where you need to define different sliding window by blocks, we can go with 1.

Reply:
1.  done
2.  forget about it
3. good suggestion, the window size actually could be different in train and test phase. But it need a smart way to implement it. any  idea? since all model parameters in decoding phase is read in model.json

Reply:
It was misleading to point out `asr_recog.py`, I'm okay to keep the same values for training and inference for now. I'll think about it.
What I wanted to highlight is that if you define `enc_win_left` and `enc_win_right` as transformer block parameters, you can change values between blocks. If it's intended, let's keep it, otherwise it should be defined as a "global" parameter (in `e2e_asr_transducer` for training and `asr_recog` for inference)

Reply:
I thinks these parameters could be different in each block. 
Yes, intended.